src/main/java/de/metas/ui/web/order/sales/purchasePlanning/view/PurchaseRowFactory.java

```diff
@@ -78,6 +83,18 @@ private PurchaseRow buildRowFromPurchaseCandidate(
 
 		final int processedPurchaseCandidateId = purchaseCandidate.isProcessed() ? purchaseCandidate.getPurchaseCandidateId() : 0;
 
+		final AvailableToPromiseQueryBuilder atpQueryBuilder = AvailableToPromiseQuery.builder();
+
+		atpQueryBuilder.productId(purchaseCandidate.getProductId());
+		atpQueryBuilder.bpartnerId(bpartnerId);
+		atpQueryBuilder.date(datePromised);
+
+		final AvailableToPromiseResult availableStockResult = new AvailableToPromiseRepository().retrieveAvailableStock(atpQueryBuilder.build());
+
+		availableStockResult.getResultGroups().get(0).getQty();
+
+		final BigDecimal availableQty = availableStockResult.getResultGroups().get(0).getQty();
```

Comment:
in AvailableToPromiseResult, i would introduce getSingleQty() which would get the single qty from result groups.